sw/device/lib/testing/test_framework/check.h

```diff
@@ -209,4 +210,24 @@
     }                                                                \
   } while (false)
 
+/**
+ * Checks that the `status_t` represents a non-error value.
+ *
+ * Prints a human-readable error message if the status represents an error.
+ *
+ * @param expr An expression which evaluates to a `status_t`.
+ */
+#define CHECK_STATUS_OK(expr, ...)                  \
+  do {                                              \
+    status_t status_ = expr;                        \
+    if (!status_ok(status_)) {                      \
+      char mod_[3];                                 \
+      int32_t arg_;                                 \
+      const char *code_;                            \
+      status_extract(status_, &code_, &arg_, mod_); \
```

Comment:
I don't think you need these local vars and the call to `status_extract` if you're using `%r` in the format string.

Reply:
Whoops, thanks for catching!